### PR TITLE
doc: update quickstart and migration guide

### PR DESCRIPTION
**Pull request by tadelesh**
- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: documentation/MIGRATION_GUIDE.md [hunk 1]
```diff
@@ -105,13 +109,16 @@ In the latest version, if a request is a long-running operation, the function na
 future, err := virtualMachinesClient.CreateOrUpdate(context.Background(), "<resource group name>", "<virtual machine name>", param)
 if err != nil {
 	log.Fatal(err)
+    return
```
Comment (ArcturusZhang) [inline documentation/MIGRATION_GUIDE.md:112]: Here is an indent inconsistency.
And do we have to add the `return` here? The `log.Fatal` will exit the program therefore the return here is unreachable

File: documentation/MIGRATION_GUIDE.md [hunk 2]
```diff
@@ -105,13 +109,16 @@ In the latest version, if a request is a long-running operation, the function na
...
 }
 if err := future.WaitForCompletionRef(context.Background(), virtualMachinesClient.Client); err != nil {
 	log.Fatal(err)
+    return
```
Comment (ArcturusZhang) [inline documentation/MIGRATION_GUIDE.md:116]: Indent issue

File: documentation/MIGRATION_GUIDE.md [hunk 3]
```diff
@@ -105,13 +109,16 @@ In the latest version, if a request is a long-running operation, the function na
...
 }
 vm, err := future.Result(virtualMachinesClient)
 if err != nil {
 	log.Fatal(err)
+    return
```
Comment (ArcturusZhang) [inline documentation/MIGRATION_GUIDE.md:121]: Indent issue

File: documentation/MIGRATION_GUIDE.md [hunk 4]
```diff
@@ -122,10 +129,12 @@ log.Printf("virtual machine ID: %v", *vm.ID)
 poller, err := client.BeginCreateOrUpdate(context.Background(), "<resource group name>", "<virtual machine name>", param, nil)
 if err != nil {
 	log.Fatal(err)
+    return
```
Comment (ArcturusZhang) [inline documentation/MIGRATION_GUIDE.md:132]: Indent issue

File: documentation/new-version-quickstart.md [hunk 5]
```diff
@@ -207,43 +210,59 @@ Example: Managing Resource Groups
...
 ***Delete a resource group***
 
 ```go
 func deleteResourceGroup(ctx context.Context, credential azcore.TokenCredential) error {
-    rgClient := armresources.NewResourceGroupsClient(subscriptionId, credential, nil)
-    
+    rgClient, err := armresources.NewResourceGroupsClient(subscriptionId, credential, nil)
+    if err != nil {
+		return err
+	}
```
Comment (ArcturusZhang) [inline documentation/new-version-quickstart.md:264]: Indent issue

File: documentation/new-version-quickstart.md [hunk 6]
```diff
@@ -259,31 +278,36 @@ func main() {
 	cred, err := azidentity.NewDefaultAzureCredential(nil)
 	if err != nil {
 		log.Fatalf("authentication failure: %+v", err)
+        return
```
Comment (ArcturusZhang) [inline documentation/new-version-quickstart.md:281]: Indent issue

File: documentation/MIGRATION_GUIDE.md (hunk 1 above)
Comment (tadelesh) [reply documentation/MIGRATION_GUIDE.md:112]: Oh, I make a mistake for the function of log.Fatel. I'll remove all the return then.

File: documentation/new-version-quickstart.md [hunk 7]
```diff
@@ -172,14 +172,19 @@ var (
...
+	}
 
-    param := armresources.ResourceGroup{
-        Location: to.StringPtr(location),
-    }
+	param := armresources.ResourceGroup{
+		Location: to.Ptr(location),
+	}
+
+	resp, err := rgClient.CreateOrUpdate(context.Background(), resourceGroupName, param, nil)
```
Comment (jhendrixMSFT) [inline documentation/new-version-quickstart.md:185]: Another case of replace `context.Background()` with `context.TODO()`.  Can be in a separate PR as there might be a more general scrub that needs to happen.

File: documentation/MIGRATION_GUIDE.md [hunk 8]
```diff
@@ -55,42 +56,32 @@ For detailed information on the benefits of using the new authentication types,
...
+- When there is an error in the SDK request, in the previous version (`services/**/mgmt/**`), the return value will all be non-nil, and you can get the raw HTTP response from the response value. In the latest version (`sdk/resourcemanager/**/arm**`), the first return value will be empty and you need to convert the error to the `azcore.ResponseError` interface to get the raw HTTP response. When the request is successful and there is no error returned, you can get the raw HTTP response from request context.
 
 **Previous version (`services/**/mgmt/**`)**
 
 ```go
 resp, err := resourceGroupsClient.CreateOrUpdate(context.Background(), resourceGroupName, resourceGroupParameters)
 if err != nil {
-	log.Fatalf("Status code: %d", resp.Response.Response.StatusCode)
+    log.Fatalf("Status code: %d", resp.Response.Response.StatusCode)
```
Comment (jhendrixMSFT) [inline documentation/MIGRATION_GUIDE.md:66]: Needs an `errors.As()` to convert to `*azcore.ResponseError` first.

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
One example needs to be updated.

File: documentation/MIGRATION_GUIDE.md (hunk 8 above)
Comment (tadelesh) [reply documentation/MIGRATION_GUIDE.md:66]: This part of code uses track1 and the raw response can be fetched by operation resp directly.

**Review by ArcturusZhang — APPROVED** [review]
